src/Confluent.Kafka/Serialization/LongDeserializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+
+using System;
+
+namespace Confluent.Kafka.Serialization
+{
+    /// <summary>
+    /// Derializer for Int64 data
+    /// </summary>
+    public class LongDeserializer : IDeserializer<long>
+    {
+        /// <summary>
+        /// Deserialize data to Int64
+        /// </summary>
+        /// <param name="data">input data</param>
+        /// <returns>deserialize result</returns>
+        public long Deserialize(byte[] data)
+        {
+            if (data == null)
+            {
+                // in accordance with java client.
+                return default(long);
```

Comment:
It's more due to language (generics can't be primitive in java so always nullable, that was easier for them and they they didn't have the choice ) and C# (where people except to find primitive serializer)

Also (but that's my point of view) I don't like the fact that it's the same method for tombstone and simple null serialization (hence the thread here : https://confluentcommunity.slack.com/archives/C499EFQS0/p1489662514821342)
People at kafka obviously don't like it either (https://cwiki.apache.org/confluence/display/KAFKA/KIP-82+-+Add+Record+Headers and others KIP)

If we want to stick with the java producer, we should force the typed consumer/producer with reference type` where TKey : class where TValue : class` so we have always null available, but we should benchmark it carefully for primitive types (int&long must be some of the most used serializer with String and Avro)